doc/release/3.2.0.rst

```diff
@@ -0,0 +1,301 @@
+Tarantool 3.2
+=============
+
+Release date: August 21, 2024
+
+Releases on GitHub: :tarantool-release:`3.2.0`
+
+The 3.2 release of Tarantool adds the following main product features and improvements for the Community and Enterprise editions:
+
+*   **Community Edition (CE)**
+
+    *   A new experimental module for validating role configurations.
+    *   Initial support for encoding structured data using Protobuf.
+    *   Next and Previous prefix iterators.
+    *   Support for all UUID versions.
+    *   Automatic loading of the most often used built-in modules into the console environment.
+
+*   **Enterprise Edition (EE)**
+
+    *   Time-to-live (TTL) for keys in a Tarantool-based configuration storage.
+
+
+
+.. _3-2-features-for-developers:
+
+Developing applications
+-----------------------
+
+.. _3-2-configuration-validation:
+
+Configuration validation
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tarantool 3.2 includes a new experimental module for :ref:`validating role configurations <roles_create_custom_role_validate>` using a declarative schema.
+For example, you can validate the type of configuration values, provide an array of allowed values, or specify a custom validation function.
+
+Suppose, a sample :ref:`'http-api' custom role <roles_example_custom_role_http_api>` can accept the ``host`` and ``port`` configuration values:
+
+..  code-block:: yaml
+
+    roles: [ http-api ]
+    roles_cfg:
+      http-api:
+        host: '127.0.0.1'
+        port: 8080
+
+First, you need to load the ``experimental.config.utils.schema`` module:
+
+..  code-block:: lua
+
+    local schema = require('experimental.config.utils.schema')
+
+The ``validate_port()`` function can be used to check that a port value is between ``1`` and ``65535``:
+
+..  code-block:: lua
+
+    local function validate_port(port, w)
+        if (port <= 1 or port >= 65535) then
+            w.error("'port' should be between 1 and 65535, got %q", port)
+        end
+    end
+
+Then, you can create a schema used for validation:
+
+-   ``host`` should be a string value from the ``allowed_values`` array.
+-   ``port`` should be a number that is checked using the ``validate_port()`` function declared above.
+
+..  code-block:: lua
+
+    local listen_address_schema = schema.new('listen_address', schema.record({
+        host = schema.scalar({
+            type = 'string',
+            allowed_values = { '127.0.0.1', '0.0.0.0' },
+        }),
+        port = schema.scalar({
+            type = 'number',
```

Comment:
Nit: `number` accepts numbers with fractional parts. I suggest to use `integer` instead.